autoload/lsc/reference.vim

```diff
@@ -138,30 +138,40 @@ function! s:showHover(result) abort
     let contents = [contents]
   endif
   let lines = []
+  let filetype = 'markdown'
   for item in contents
     if type(item) == type({})
       let l:lines += split(item.value, "\n")
+      if has_key(item, 'language')
+        let l:filetype = item.language
+      elseif has_key(item, 'kind')
+        let l:filetype = item.kind
```

Comment:
This should only ever be `'markdown'` or `'plaintext'`, but setting `filetype` to `'plaintext'` is unlikely to be useful. How about `let l:filetype = item.kind == 'markdown' ? 'markdown' : 'text'`

Reply:
Hello Nate,

The TSServer Language Server sets the file type to TypeScript, not Text or Markdown. I don’t know the LSP spec, maybe that should not happen, but it does. The vim-lsp plugin handles this, that’s where I got the logic from.

So I do believe we need to handle that particular case.

But most language servers will provide either Text or Markdown.

Reply:
That should happen with the `'language'` key (the value is a `MarkedString`), but never with the `'kind'` key (the value is a `MarkeupContent`).

We can definitely keep the `let l:filetype = item.language` line, it's only the handling of 'kind' that I think should change.

Does TSServer use 'kind' or 'language'?

Reply:
Of the top of my head, I can't remember.

I will get back to you on this Nate, hopefully soon.

Reply:
The [TypeScript & JavaScript Language Server](https://github.com/theia-ide/typescript-language-server) indeed sets `language` not `kind` as noted here by this example snippet I just ran:

```
[{'language': 'typescript', 'value': '(method) AxiosInstance<any>.....', ''}]
```

So your suggestion seems correct, the code should be:

```viml
  for item in contents
    if type(item) == type({})
      let l:lines += split(item.value, "\n")
      if has_key(item, 'language')
        let l:filetype = item.language
      elseif has_key(item, 'kind')
        let l:filetype = item.kind == 'markdown' ? 'markdown' : 'text'
      endif
    else
      let l:lines += split(item, "\n")
    endif
  endfor
```

I will push that change.